client/components/Coverages/CoverageEditor/CoverageForm.jsx

```diff
@@ -14,156 +14,174 @@ import {
     Field,
 } from '../../UI/Form';
 
-export const CoverageForm = ({
-    field,
-    value,
-    onChange,
-    newsCoverageStatus,
-    dateFormat,
-    timeFormat,
-    contentTypes,
-    genres,
-    keywords,
-    readOnly,
-    item,
-    diff,
-    formProfile,
-    errors,
-    showErrors,
-    currentWorkspace,
-}) => {
-    const onScheduleChanged = (f, v) => {
-        if (f.endsWith('.date')) {
-            // If there is no current scheduled date, then set the time value to end of the day
-            if (!get(value, 'planning.scheduled')) {
-                onChange(
-                    f.slice(0, -5),
-                    v.endOf('day')
-                );
-            } else {
-                onChange(f.slice(0, -5), v);
-            }
-        } else if (f.endsWith('.time')) {
-            // If there is no current scheduled date, then set the date to today
-            if (!get(value, 'planning.scheduled')) {
-                onChange(
-                    f.slice(0, -5),
-                    moment()
-                        .hour(v.hour())
-                        .minute(v.minute())
-                );
+export class CoverageForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.dom = {contentType: null};
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.hasAssignment && this.props.hasAssignment) {
+            this.dom.contentType.focus();
+        }
+    }
+
+    render() {
+        const {
+            field,
+            value,
+            onChange,
+            newsCoverageStatus,
+            dateFormat,
+            timeFormat,
+            contentTypes,
+            genres,
+            keywords,
+            readOnly,
+            item,
+            diff,
+            formProfile,
+            errors,
+            showErrors,
+            currentWorkspace,
+            hasAssignment,
+        } = this.props;
+
+        const onScheduleChanged = (f, v) => {
```

Comment:
This can now be changed to be a method of the class itself rather than an anonymous function of the render method